llvm/lib/Target/NVPTX/NVPTXAsmPrinter.cpp

```diff
@@ -1682,29 +1673,37 @@ void NVPTXAsmPrinter::bufferLEByte(const Constant *CPV, int Bytes,
 void NVPTXAsmPrinter::bufferAggregateConstant(const Constant *CPV,
                                               AggBuffer *aggBuffer) {
   const DataLayout &DL = getDataLayout();
-  int Bytes;
 
-  // Integers of arbitrary width
-  if (const ConstantInt *CI = dyn_cast<ConstantInt>(CPV)) {
-    APInt Val = CI->getValue();
+  auto BufferConstant = [&](APInt Val) {
```

Comment:
I'd rename it `ExtendBuffer` and pass aggBuffer explicitly. Makes it easier to see what's going on.

Reply:
Renamed.